app/BUILD.bazel

```diff
@@ -128,72 +167,84 @@ VIEW_MODEL_FILES = [
-    "src/main/java/org/oppia/app/player/state/listener/NextNavigationButtonListener.kt",
-    "src/main/java/org/oppia/app/player/state/listener/PreviousNavigationButtonListener.kt",
-    "src/main/java/org/oppia/app/player/state/listener/PreviousResponsesHeaderClickListener.kt",
-    "src/main/java/org/oppia/app/player/state/listener/ReplayButtonListener.kt",
-    "src/main/java/org/oppia/app/player/state/listener/ReturnToTopicNavigationButtonListener.kt",
-    "src/main/java/org/oppia/app/player/state/listener/RouteToHintsAndSolutionListener.kt",
-    "src/main/java/org/oppia/app/player/state/listener/StateKeyboardButtonListener.kt",
-    "src/main/java/org/oppia/app/player/state/listener/SubmitNavigationButtonListener.kt",
-    "src/main/java/org/oppia/app/profile/RouteToAdminPinListener.kt",
-    "src/main/java/org/oppia/app/profileprogress/ProfilePictureClickListener.kt",
-    "src/main/java/org/oppia/app/profileprogress/RouteToCompletedStoryListListener.kt",
-    "src/main/java/org/oppia/app/profileprogress/RouteToOngoingTopicListListener.kt",
-    "src/main/java/org/oppia/app/recyclerview/OnDragEndedListener.kt",
-    "src/main/java/org/oppia/app/recyclerview/OnItemDragListener.kt",
-    "src/main/java/org/oppia/app/story/ExplorationSelectionListener.kt",
-    "src/main/java/org/oppia/app/topic/lessons/StorySummarySelector.kt",
-    "src/main/java/org/oppia/app/topic/revision/RevisionSubtopicSelector.kt",
-    "src/main/java/org/oppia/app/topic/revisioncard/ReturnToTopicClickListener.kt",
-    "src/main/java/org/oppia/app/topic/RouteToRevisionCardListener.kt",
-    "src/main/java/org/oppia/app/utility/OnClickableAreaClickedListener.kt",
-    "src/main/java/org/oppia/app/utility/RegionClickEvent.kt",
+# TODO(#1617): Remove genrules post-gradle
+# Files altered by genrules to be excluded in the app library
+ALTERED_VIEW_MODELS = [
+    "src/main/java/org/oppia/app/administratorcontrols/administratorcontrolsitemviewmodel/AdministratorControlsAccountActionsViewModel.kt",
+    "src/main/java/org/oppia/app/help/faq/FAQListViewModel.kt",
+    "src/main/java/org/oppia/app/help/HelpItemViewModel.kt",
+    "src/main/java/org/oppia/app/help/HelpListViewModel.kt",
+    "src/main/java/org/oppia/app/onboarding/OnboadingSlideViewModel.kt",
+    "src/main/java/org/oppia/app/onboarding/OnboardingViewModel.kt",
+    "src/main/java/org/oppia/app/parser/StringToFractionParser.kt",
+    "src/main/java/org/oppia/app/parser/StringToNumberParser.kt",
+    "src/main/java/org/oppia/app/player/state/itemviewmodel/FractionInteractionViewModel.kt",
+    "src/main/java/org/oppia/app/player/state/itemviewmodel/ImageRegionSelectionInteractionViewModel.kt",
+    "src/main/java/org/oppia/app/topic/info/TopicInfoViewModel.kt",
 ]
 
-ANNOTATION_FILES = [
-    "src/main/java/org/oppia/app/activity/ActivityScope.kt",
-    "src/main/java/org/oppia/app/fragment/FragmentScope.kt",
-    "src/main/java/org/oppia/app/utility/KeyboardHelper.kt",
-]
+[genrule(
```

Comment:
Please add comment explaining what this genrule is doing & why it's needed.

Reply:
Done!